book/src/developers/contributing/build_instructions/mac_os.md

```diff
@@ -1,9 +1,98 @@
 # Mac Os
 
-Clone trin and run.
+## Prerequisites
+- [Rust](https://www.rust-lang.org/) installation
+- Xcode Command Line Tools
+
+## Building, Testing, and Running
+
+Note: If you use a VPN, you should disable it before running Trin.
+
+Install Xcode Command Line Tools (MacOS):
+
+```sh
+xcode-select --install
+```
+
+Add environment variables in current shell session:
+
+```sh
+# Optional
+export RUST_LOG=<error/warn/info/debug/trace>
+export TRIN_DATA_PATH=<path-to-data-directory>
+```
+
+Install Rust dependencies (MacOS):
+
+```sh
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs/ | sh -s -- --default-toolchain stable -y
+
+# Permanently add environment variables to login shell file for reuse across terminal sessions.
+if [[ $(echo $SHELL) == "/bin/zsh" ]]; then
+  LOGIN_SHELL_FILE=$HOME/.zshrc
+elif [[ $(echo $SHELL) == "/bin/bash" ]]; then
+  LOGIN_SHELL_FILE=$HOME/.bashrc
+else
+  echo "The path to the binary of your current shell is: $SHELL"
+  echo "Please manually add environment variables to the relevant login shell file" 
+  exit 1
+fi
+echo 'export RUST_LOG=debug' >> $LOGIN_SHELL_FILE
```

Comment:
Setting ``export RUST_LOG=debug`` shouldn't be here. Trin has default logging options, if a user wants to set logging options they can themselves